### PR TITLE
Feature - Relations To Array

### PR DESCRIPTION
**Pull request by jfo84**
There's a lot of changes here. The new `to` array feature itself still needs testing.

So, the array approach is much more performant. And it doesn't require using currying techniques to allow one DOM element to draw arrows to many DOM elements.

Most of the ArcherElement test is rewritten to use a mock ArcherContainer. Test coverage is way up.

These are obviously breaking changes and will require a major semver. To be clear, I'm not going to try very hard to merge this if you want major changes to the PR. We like this approach a lot more than the old one.

Let me know @pierpo

### CODE REVIEW

**Comment by pierpo** [comment]
Hi @jfo84! Thanks for the PR :wink: 
Indeed, that's quite a lot of changes!

Can you update the example provided in the repo so that I have a clearer view on the API changes?

I'll have a deeper look into this when I have more time but it feels right!

(be careful though, you took your 2 other commits from the other branch with you, and your last commit has a very strange name :smile: )

**Comment by jfo84** [comment]
I added some snapshot testing for `computeArrows` and `generateAllArrowMarkers`. I wish we could have standalone files for the snapshot tests but they need the unique prefix on each run

That should be it for changes

**Comment by jfo84** [comment]
Third example added with the new syntax. That should be it @pierpo

File: src/ArcherContainer.js [hunk 1]
```diff
@@ -306,6 +330,27 @@ export class ArcherContainer extends React.Component<Props, State> {
 
       const arrowPath = `M0,0 L0,${arrowThickness} L${arrowLength -
         1},${arrowThickness / 2} z`;
+      
+      if (Array.isArray(to)) {
+        return (
+          <React.Fragment key={JSON.stringify({ from, to })}>
+            {to.map((relationNozzle: RelationNozzleType) => (
+              <marker
```
Comment (pierpo) [inline src/ArcherContainer.js]: Maybe we should refactor this `marker` element, right? Since it appears twice.

**Comment by pierpo** [comment]
@jfo84 Amazing job, congrats. I really like this new API, it makes way more sense. Thank you.

I left a comment in the code and I have two general questions:

- You said it would break the old API but since the old examples still work I can't think of a case where it breaks. Any simple breaking example?
- The `label` feature is not implemented 😉 I guess we need a small API change here too, but I don't know how to make it consistent


Before:

```javascript
{
  from: { ... }
  to: { ... }
  label: 'my label'
}
```

Now would need to be:

```javascript
{
  from: { ... }
  to: [{ label: 'my label', ... }]
}
```
and

```javascript
{
  from: { ... }
  to: { label: 'my label', ... }
}
```

I wonder if we should keep the old way. I think we shouldn't.

**Comment by jfo84** [comment]
Hey @pierpo I essentially re-wrote it. I'm going to put up another PR for clarity. This one removes a lot of the nasty merging code

**Pull request closed without merging**